src/math/CMakeLists.txt

```diff
@@ -5,31 +5,41 @@ if(BUILD_LIBRARY)
 	return()
 endif()
 
+is_zephyr(it_is_zephyr)
+
 add_local_sources(sof numbers.c)
 
-if(CONFIG_CORDIC_FIXED)
+if(CONFIG_CORDIC_FIXED OR it_is_zephyr)
         add_local_sources(sof trig.c)
 endif()
 
 add_local_sources_ifdef(CONFIG_SQRT_FIXED sof sqrt_int16.c)
 
 add_local_sources_ifdef(CONFIG_MATH_EXP sof exp_fcn.c exp_fcn_hifi.c)
 
-if(CONFIG_MATH_DECIBELS)
+if(CONFIG_MATH_DECIBELS OR it_is_zephyr)
         add_local_sources(sof decibels.c)
 endif()
 
+# Never used with Zephyr.
+if(NOT it_is_zephyr)
 add_local_sources_ifdef(CONFIG_NATURAL_LOGARITHM_FIXED sof log_e.c)
 
 add_local_sources_ifdef(CONFIG_COMMON_LOGARITHM_FIXED sof log_10.c)
 
 add_local_sources_ifdef(CONFIG_POWER_FIXED sof power.c)
 
 add_local_sources_ifdef(CONFIG_BINARY_LOGARITHM_FIXED sof base2log.c)
+endif()
 
+if(it_is_zephyr)
+add_local_sources_ifdef(CONFIG_COMP_FIR sof fir_generic.c fir_hifi2ep.c fir_hifi3.c)
+else()
 add_local_sources_ifdef(CONFIG_MATH_FIR sof fir_generic.c fir_hifi2ep.c fir_hifi3.c)
+endif()
 
-if(CONFIG_MATH_FFT)
+# Never used with Zephyr.
+if(CONFIG_MATH_FFT AND NOT it_is_zephyr)
```

Comment:
actually this would be called in MFCC, as well as window and dct, though MFCC is not included in zephyr now, but can we keep as original?

Reply:
You lost me sorry: what is different from the original? I tried really hard not to change anything.

Reply:
okay, since the dependency of MFCC libs haven't been included in zephyr yet, it is okay to modify code latter if we want to add mfcc to zephyr.